1-Parallelize-your-python-code_Futures_and_Delayed.ipynb

```diff
@@ -0,0 +1,1062 @@
+    "tag = \"dask\"\n",
+    "query_filter = \"Newest\"\n",
+    "url = f\"{base_url}{tag}?tab={query_filter}\"\n",
+    "url"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c19431d0-8e9a-42f2-8ed7-30994ed6986a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = requests.get(url)\n",
+    "html_str = r.text\n",
+    "html = HTML(html=html_str)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eea631c7-76b1-4b26-aaee-ef42f2f472ba",
+   "metadata": {},
+   "source": [
+    "`.s-post-summary`: Base parent container for a post summary\n",
+    "\n",
+    "https://stackoverflow.design/product/components/post-summary/\n",
+    "\n",
+    "`html.find(\".s-post-summary\")` returns list of 50 elements, let's explore one of them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7dc749f0-c7ea-44bf-aa9c-a8569d742b12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "html.find(\".s-post-summary\")[0].text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ecb88c30-e6ea-426b-9322-bf9470de05ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(html.find(\".s-post-summary\")[0].text)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a514c55c-40ba-496c-8211-a99e01832221",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "question = html.find(\".s-post-summary\")[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39b1a8aa-9f23-4f3e-b3db-5396ed72c9ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "question.text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e5d96971-7404-49c8-9d10-b5659e36badd",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "question.find(\".s-post-summary--content-title\", first=True).text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e523217d-7ff3-4d47-9f24-d5c2ac461c1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "question.find(\".s-post-summary--stats\", first=True).text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e848e39a-7daf-4c20-ac0d-feae561b7c89",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "question.find(\".s-post-summary--meta-tags\", first=True).text"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d22c3321-d821-4c89-b90d-a1620312db0b",
+   "metadata": {},
+   "source": [
+    "## Scrape and clean \n",
+    "\n",
+    "With this information, we wrote some functions, that get a page and clean the data from all the posts in that page and returns it as a list of dictionaries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26ea6fc0-74da-4c51-ab04-975ac51aed31",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def extract_url_and_parse_html(url):\n",
```

Comment:
You might consider splitting up the `extract_url_and_parse_html` function a bit.  This would force the students to chain things with submit or map and think a bit about dependencies.  It will also make the dashboard more interesting I think.  One way to do this would be to bring `requests.get` outside of the function and call it separately